src/components/core/geoChart/GeoChartLegendRenderer.tsx

```diff
@@ -1,12 +1,216 @@
-// (C) 2007-2019 GoodData Corporation
+// (C) 2020 GoodData Corporation
 import * as React from "react";
+import without = require("lodash/without");
+import * as classNames from "classnames";
+import { Execution } from "@gooddata/typings";
+import { IntlWrapper } from "../../core/base/IntlWrapper";
+import { IntlTranslationsProvider, ITranslationsComponentProps } from "../../core/base/TranslationsProvider";
+import { stringToFloat } from "../../../helpers/utils";
+import PushpinSizeLegend from "./legends/PushpinSizeLegend";
+import { getGeoDataIndex } from "../../../helpers/geoChart";
+import { IGeoConfig, IGeoDataIndex } from "../../../interfaces/GeoChart";
+import { COLOR, SIZE, SEGMENT_BY } from "../../../constants/bucketNames";
+import { TOP } from "../../visualizations/chart/legend/PositionTypes";
+import Legend, { ILegendProps } from "../../visualizations/chart/legend/Legend";
```

Comment:
should we re-use HighCharts's Legend? @naibinh

Reply:
We can extract it to a common place for both Highchart and Mapbox legends :)